editor/editor_properties.cpp

```diff
@@ -1801,6 +1862,27 @@ void EditorPropertyVector3::_value_changed(double val, const String &p_name) {
 
 void EditorPropertyVector3::update_property() {
 	update_using_vector(get_edited_object()->get(get_edited_property()));
+	_update_ratio();
+}
+
+void EditorPropertyVector3::_update_ratio() {
+	linked->set_modulate(Color(1, 1, 1, linked->is_pressed() ? 1.0 : 0.5));
+
+	if (spin[0]->get_value() != 0 && spin[1]->get_value() != 0) {
+		ratio_yx = spin[1]->get_value() / spin[0]->get_value();
+		ratio_zx = spin[2]->get_value() / spin[0]->get_value();
+		ratio_xy = spin[0]->get_value() / spin[1]->get_value();
+		ratio_zy = spin[2]->get_value() / spin[1]->get_value();
+		ratio_xz = spin[0]->get_value() / spin[2]->get_value();
+		ratio_yz = spin[1]->get_value() / spin[2]->get_value();
+	} else {
+		ratio_yx = 1.0;
+		ratio_zx = 1.0;
+		ratio_xy = 1.0;
+		ratio_zy = 1.0;
+		ratio_xz = 1.0;
+		ratio_yz = 1.0;
+	}
```

Comment:
`ratio_yx` is the same as `1.0 / ratio_xy`.
Maybe you could reduce the amount of member variables and computation needed by keeping only 3 and using `*` or `/` operations as appropriate?

Reply:
I could, I even tried it at first, but IIRC there were some precision issues.